examples/02-javascript-basics/array_manipulation_library.js

```diff
@@ -11,12 +11,50 @@ Join - join the array elements into a string given a parameter
 */
 
 const $ = {
-	filter: function () {},
-	map: function () {},
-	reduce: function () {},
-	head: function () {},
-	tail: function () {},
-	join: function () {}
+	filter: function (arr, fn) {
+    let newArr = [];
+    for (i = 0; i < arr.length; i++) {
+      if (fn(arr[i])) newArr.push(arr[i]);
+    }
+    return newArr;
+  },
+	map: function (arr, fn) {
```

Comment:
@anneilustre you might need to adjust your editor, it's causing some indentations